torchsummary/torchsummary.py

```diff
@@ -34,7 +34,10 @@ def hook(module, input, output):
                not (module == model)):
                 hooks.append(module.register_forward_hook(hook))
                 
-        if torch.cuda.is_available():
+        device = device.lower()
+        assert device in ["cuda", "cpu"], "Input device is not valid, please specify 'cuda' or 'cpu'"
+
+        if device is "cuda" and torch.cuda.is_available():
```

Comment:
- Since you used `device = device.lower()` on L37, you can no longer use `is` for comparing string. 
- Now `device` and `"cuda"` have different `ids`, therefore `==` should be used.